### PR TITLE
Add ESPHome 2024.6.0 compatibility

### PR DESCRIPTION
**Pull request by Roving-Ronin**
- Added ota "platform" to all (required from ESPHome 2024.6.0)
- Added ESPHome "min_version" to all and adjusted existing to "2024.6.0" 
- Updated minimum version required to '2024.6.0 or greater' in the README.md

### CODE REVIEW

**Review by syssi — APPROVED** [review]
Could you update the minimum required ESPHome version at the README.md too?

**Comment by Roving-Ronin** [comment]
Minimum required ESPHome version has been updated in the README.md

File: README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ All Victron devices providing a ve.direct port.
 
 ## Requirements
 
-* [ESPHome 2021.10 or higher](https://github.com/esphome/esphome/releases).
+* [ESPHome 2026.6.0 or higher](https://github.com/esphome/esphome/releases).
```
Comment (syssi) [inline README.md:27]:
```suggestion
* [ESPHome 2024.6.0 or higher](https://github.com/esphome/esphome/releases).
```

File: smartsolar-mppt-esp8266-example-advanced.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ substitutions:
   external_components_source: github://KinDR007/VictronMPPT-ESPHOME@main
   lower_devicename: "victron"
   devicename: "SmartSolar_150_35"
-  config_version: "v2021.05.30"
+  config_version: "v2021.06.21"
```
Comment (syssi) [inline smartsolar-mppt-esp8266-example-advanced.yaml:5]:
```suggestion
  config_version: "v2024.06.21"
```

**Comment by Roving-Ronin** [comment]
Sorry my bad (typo), overlooked that 2021 --> 2024 update.
IMHO all these examples need to be updated to a consistent format also, at the moment they are all different.

Helping a friend setup a Victron MultiPlus-II 12/3000 and MPPT 250/70 units with Athom M5 Lite (ESP32) shortly, will see about uploading those as new example templates.

**Comment by syssi** [comment]
Thanks for your contribution! Feel free to provide more examples and improvements!

**Pull request merged**